pandas/tests/indexes/categorical/test_category.py

```diff
@@ -353,16 +353,78 @@ def test_is_monotonic(self, data, non_lexsorted_data):
         assert c.is_monotonic_decreasing is False
 
     def test_has_duplicates(self):
-
         idx = CategoricalIndex([0, 0, 0], name="foo")
         assert idx.is_unique is False
         assert idx.has_duplicates is True
 
+        idx = CategoricalIndex([0, 1], categories=[2, 3], name="foo")
+        assert idx.is_unique is False
+        assert idx.has_duplicates is True
+
+        idx = CategoricalIndex([0, 1, 2, 3], categories=[1, 2, 3], name="foo")
+        assert idx.is_unique is True
+        assert idx.has_duplicates is False
+
+    def _test_drop_duplicates(self, idx, keep, expected, index):
+        for k, e, i in zip(keep, expected, index):
+            tm.assert_numpy_array_equal(idx.duplicated(keep=k), e)
+            e = idx[~e]
+
+            result = idx.drop_duplicates(keep=k)
+            tm.assert_index_equal(result, e)
+
+            result = Series(idx).drop_duplicates(keep=k)
```

Comment:
if you're testing Series.drop_duplicates, this should go in tests.series.methods.test_drop_duplicates

Reply:
Previous tests included this kind of test. Series already has drop_duplicates tests, so I think I can delete it here.